test/e2e/e2e_test.go

```diff
@@ -184,3 +321,46 @@ func registerBroker(namespace string) error {
 
 	return err
 }
+
+// setupInstance will create an instance for each test case in their own unique namespace
+func setupServiceInstance(t *testing.T, namespace string) (*v1beta1.ServiceInstance, error) {
+	// cluster resource definition interface for service instance
+	crd := &v1beta1.ServiceInstance{}
+	testutil.ReadInYAMLFileAndConvert("../../samples/kubernetes/instance.yaml", &crd)
+
+	// Provision instance from the CRD using the service catalog
+	provisionedCRD, err := svcatClient.ServicecatalogV1beta1().ServiceInstances(namespace).Create(crd)
+	if !assert.NoError(t, err) {
+		return nil, err
+	}
+
+	// Naming convention used in kubernetes
+	serviceinstance := &v1beta1.ServiceInstance{}
+	// Wait up to 25 minutes for the service instance to have been provisioned.
+	err = testutil.Poll(25, func() (bool, error) {
+		// Get the instance created by the service catalog
+		serviceinstance, err = svcatClient.ServicecatalogV1beta1().ServiceInstances(namespace).Get(provisionedCRD.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		return v1beta1.ServiceInstanceProvisionStatus("Provisioned") == serviceinstance.Status.ProvisionStatus, nil
+	})
+
+	return serviceinstance, err
+}
+
+// setupServiceBinding will create a binding for each test case in their own unique namespace
+func setupServiceBinding(t *testing.T, namespace string) (*v1beta1.ServiceBinding, error) {
+	// cluster resource definition interface for service binding
+	crd := &v1beta1.ServiceBinding{}
+	testutil.ReadInYAMLFileAndConvert("../../samples/kubernetes/binding.yaml", &crd)
+
+	// Bind user to service instance
+	servicebinding, err := svcatClient.ServicecatalogV1beta1().ServiceBindings(namespace).Create(crd)
```

Comment:
is there any status for binding we could check as well? May be no error could be returned but it could be in some "Pending" status or whatever